server/routes/archiveGoal/archiveGoalController.test.ts

```diff
@@ -0,0 +1,258 @@
+        reason: ReasonToArchiveGoalValue.PRISONER_NO_LONGER_WANTS_TO_WORK_TOWARDS_GOAL,
+      }
+      const expectedView = {
+        prisonerSummary,
+        form: expectedForm,
+      }
+      req.session.archiveGoalForm = expectedForm
+
+      // When
+      await controller.getReviewArchiveGoalView(
+        req as undefined as Request,
+        res as undefined as Response,
+        next as undefined as NextFunction,
+      )
+
+      // Then
+      expect(res.render).toHaveBeenCalledWith('pages/goal/archive/review', expectedView)
+    })
+  })
+
+  describe('getArchiveGoalReviewForm', () => {
+    it('should request the goal is archived', async () => {
+      // Given
+      req.user.token = 'some-token'
+      req.session.archiveGoalForm = {
+        reference: goalReference,
+        title: 'Some goal',
+        reason: ReasonToArchiveGoalValue.PRISONER_NO_LONGER_WANTS_TO_WORK_TOWARDS_GOAL,
+      }
+      const expectedArchiveGoalDto: ArchiveGoalDto = {
+        prisonNumber,
+        goalReference,
+        reason: ReasonToArchiveGoalValue.PRISONER_NO_LONGER_WANTS_TO_WORK_TOWARDS_GOAL,
+      }
+      mockedArchiveGoalFormToArchiveGoalDtoMapper.mockReturnValue(expectedArchiveGoalDto)
+
+      // When
+      await controller.submitReviewArchiveGoal(
+        req as undefined as Request,
+        res as undefined as Response,
+        next as undefined as NextFunction,
+      )
+
+      // Then
+      expect(educationAndWorkPlanService.archiveGoal).toHaveBeenCalledWith(expectedArchiveGoalDto, 'some-token')
+      expect(res.redirect).toHaveBeenCalledWith(`/plan/${prisonNumber}/view/overview`)
+      expect(req.session.archiveGoalForm).toBeUndefined()
+    })
+    it('should handle a failure archiving the goal', async () => {
+      // Given
+      req.user.token = 'some-token'
+      req.session.archiveGoalForm = {
+        reference: goalReference,
+        title: 'Some goal',
+        reason: ReasonToArchiveGoalValue.PRISONER_NO_LONGER_WANTS_TO_WORK_TOWARDS_GOAL,
+      }
+      const expectedArchiveGoalDto: ArchiveGoalDto = {
+        prisonNumber,
+        goalReference,
+        reason: ReasonToArchiveGoalValue.PRISONER_NO_LONGER_WANTS_TO_WORK_TOWARDS_GOAL,
+      }
+      mockedArchiveGoalFormToArchiveGoalDtoMapper.mockReturnValue(expectedArchiveGoalDto)
+      educationAndWorkPlanService.archiveGoal.mockRejectedValue(createError(500, 'Service unavailable'))
+      const expectedError = createError(500, `Error archiving goal for prisoner ${prisonNumber}`)
+
+      // When
+      await controller.submitReviewArchiveGoal(
+        req as undefined as Request,
+        res as undefined as Response,
+        next as undefined as NextFunction,
+      )
+
+      // Then
+      expect(educationAndWorkPlanService.archiveGoal).toHaveBeenCalledWith(expectedArchiveGoalDto, 'some-token')
+      expect(next).toHaveBeenCalledWith(expectedError)
+      expect(req.session.archiveGoalForm).toBeUndefined()
+    })
+  })
```

Comment:
Do we need a test to cover the scenario where the user has clicked No, in which case we want to assert they are redirected back to the Overview page and we have not called the API ?

Reply:
Scrap that! 😁 
I've just seen the nunjucks for this template and can see the No button is an anchor link, so doesnt actually submit the form to the controller method at all 👍

Reply:
That doesn't happen within the controller, the "no" button is just a link to overview. There is a cypress test covering that flow